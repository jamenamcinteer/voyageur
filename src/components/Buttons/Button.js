import React from "react";
import styled from "styled-components";

const StyledButton = styled.button.attrs(props => ({
  backgroundcolor:
    props.buttontype === "secondary"
      ? "#fff"
      : props.buttontype === "link"
      ? "transparent"
      : props.buttontype === "delete"
      ? "transparent"
      : props.theme.themeColor,
  border:
    props.buttontype === "secondary"
      ? `1px solid ${props.theme.themeColor}`
      : props.buttontype === "link"
      ? "0"
      : props.buttontype === "delete"
      ? "0"
      : "0"
}))`
  background-color: ${props => props.backgroundcolor};
  border: ${props => props.border};
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  width: ${props => (props.buttonwidth ? props.buttonwidth : "auto")};
  text-decoration: none;
  display: block;
  text-align: center;
  margin: 10px;
`;

const StyledButtonText = styled.span.attrs(props => ({
  color:
    props.buttontype === "secondary"
      ? props.theme.themeColor
      : props.buttontype === "link"
      ? props.theme.themeColor
      : props.buttontype === "delete"
      ? props.theme.themeColorRed
      : "#fff",
  textdecoration:
    props.buttontype === "secondary"
      ? "none"
      : props.buttontype === "link"
      ? "underline"
      : props.buttontype === "delete"
      ? "underline"
      : "none"
}))`
  font-size: 1em;
  font-weight: normal;
  color: ${props => props.color};
  text-decoration: ${props => props.textdecoration};
`;

const Button = props => {
  let customStyles = {
    background:
      props.customStyles && props.customStyles.background
        ? props.customStyles.background
        : {},
    text:
      props.customStyles && props.customStyles.text
        ? props.customStyles.text
        : {}
  };

  return (
    <StyledButton
      onClick={props.handleClick}
      buttonwidth={props.buttonWidth}
      buttontype={props.buttonType}
      style={customStyles.background}
    >
      <StyledButtonText buttontype={props.buttonType} style={customStyles.text}>
        {props.buttonText}
      </StyledButtonText>
    </StyledButton>
  );
};

export default Button;
