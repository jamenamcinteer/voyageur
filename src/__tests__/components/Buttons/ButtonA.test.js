import React from "react";
import { render, cleanup } from "@testing-library/react";
import Button from "../../../components/Buttons/ButtonA";

afterEach(cleanup);

test("should render with no props", () => {
  const { asFragment } = render(<Button />);
  expect(asFragment()).toMatchSnapshot();
});

test("should render passed props and respond to callback props", () => {
  const to = "/";
  const customStyles = {
    background: {
      backgroundColor: "orange"
    },
    text: {
      color: "teal"
    }
  };
  const buttonType = "secondary";
  const buttonWidth = "75px";
  const buttonText = "TEST";
  const { getByTestId, asFragment } = render(
    <Button
      to={to}
      customStyles={customStyles}
      buttonType={buttonType}
      buttonWidth={buttonWidth}
      buttonText={buttonText}
      id="1"
    />
  );

  expect(asFragment()).toMatchSnapshot();

  expect(getByTestId("buttonText1").textContent).toBe("TEST");
});
