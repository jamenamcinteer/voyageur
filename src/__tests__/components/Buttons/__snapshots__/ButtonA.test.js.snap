// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render passed props and respond to callback props 1`] = `
<DocumentFragment>
  <a
    class="sc-bdVaJa dmGgTf"
    data-testid="button1"
    href="/"
    style="background-color: orange;"
  >
    <span
      class="sc-bwzfXH knFLdz"
      data-testid="buttonText1"
      style="color: teal;"
    >
      TEST
    </span>
  </a>
</DocumentFragment>
`;

exports[`should render with no props 1`] = `
<DocumentFragment>
  <a
    class="sc-bdVaJa fMfgFT"
  >
    <span
      class="sc-bwzfXH hAnPLf"
      color="#fff"
    />
  </a>
</DocumentFragment>
`;
